mlir-tensorrt/compiler/lib/Dialect/Plan/Transforms/RefineTypes.cpp

```diff
@@ -294,6 +294,62 @@ struct StableHloRefineTypeFromWithShapeGeneric
   }
 };
 
+/// Given a pattern `plan.with_shape(tensorrt_op, dims...)`, if inspection of
+/// `dims` yields an opportunity to refine the type of `with_shape`, then
+/// `tensorrt_op` can also be refined. The refinements are made (and casts are
+/// inserted if required).
+struct TensorRTRefineTypeFromWithShapeGeneric
+    : public OpRewritePattern<WithShapeOp> {
+  using OpRewritePattern<WithShapeOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(WithShapeOp withOp,
+                                PatternRewriter &rewriter) const override {
+    auto producer = withOp.getOperand().getDefiningOp();
+    if (!producer || !producer->hasOneUse() ||
+        !isa<tensorrt::TensorRTDialect>(producer->getDialect()))
+      return failure();
+
+    // Create a new shape and try to refine it.
+    std::optional<SmallVector<int64_t>> newShape =
+        getRefinedShape(withOp.getShape(), withOp.getOperand().getType());
+    if (!newShape)
+      return failure();
+
+    // Update type of the producer.
+    updateTypeInPlaceAndMaybeInsertCast(
+        rewriter, withOp.getOperand(),
+        withOp.getOperand().getType().clone(*newShape));
+
+    // Update type of the WithShapeOp.
+    updateTypeInPlaceAndMaybeInsertCast(rewriter, withOp.getResult(),
+                                        withOp.getType().clone(*newShape));
+    return success();
+  }
+};
+
+struct TensorRTResizeAbsorbCastPattern
```

Comment:
The resize op can work well without this pattern, should I keep it? Also should it only apply to `tensorrt.resize` or for all ops in `tensorrt` dialect if we need it?

Reply:
No don't keep it if it's not necessary

Reply:
Removed, thanks!